XenAdmin/ConsoleView/XSVNCScreen.cs

```diff
@@ -1085,6 +1085,8 @@ private void ConnectHostedConsole(VNCGraphicsClient v, Console console)
                 throw new Failure(Failure.INTERNAL_ERROR, Messages.HOST_GONE);
             }
 
+            console.location = System.Text.RegularExpressions.Regex.Replace(console.location, @"((25[0-5]|2[0-4][0-9]|[01]?[0-9][0-9]?)(\.)){3}(25[0-5]|2[0-4][0-9]|[01]?[0-9][0-9]?){1}(:([0-6][0-5][0-5][0-3][0-5]|[0-5]\d{1,4}|\d{1,4}))?", console.Connection.HostnameWithPort);

```

Comment:
@ander089 after studying your regex I have found:

1. That some valid port numbers are excluded such as `1.1.1.1:65509` where the last digit will not be included in the match and therefore after replacement will include it twice.
2. I'm not sure if this is supported, but the regex won't handle IPv6 addresses.
3. I am struggling to think of a scenario where replacing the IP:port with Hostname:port would help make the connection more reliable to set up.
4. I can think of scenarios where the IP:port would be more reliable; differing DNS setups available to the local client computer and the server for example.